### ISSUE TITLE
`falco-builder package` fails with BUILD_BPF=ON

### ISSUE THREAD

**Issue by jalseth** [falcosecurity/falco#900]
**What happened**:

Build fails.

**What you expected to happen**:

Build succeeds.

**How to reproduce it (as minimally and precisely as possible)**:

```bash
mkdir src && cd src
git clone https://github.com/falcosecurity/falco && cd falco && git checkout tags/0.17.1 && cd ..
git clone https://github.com/draios/sysdig && cd sysdig && git checkout tags/0.26.4 && cd ..
docker run -e BUILD_BPF=ON -v $PWD:/source -v $PWD/build:/build falcosecurity/falco-builder cmake 2>&1 | tee cmake-ebpf.log
docker run -v $PWD:/source -v $PWD/build:/build falcosecurity/falco-builder package 2>&1 | tee package-ebpf.log
```

**Anything else we need to know?**:

It works fine with `BUILD_BPF=OFF`

Logs: https://gist.github.com/jalseth/3c52d944a8305d547c271295ee54f925

**Environment**:
- Falco version (use `falco --version`): 0.17.1
- Cloud provider or hardware configuration: GCP
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04.3 LTS
- Kernel (e.g. `uname -r`): 5.0.0-1021-gcp
- Install tools (e.g. in kubernetes, rpm, deb, from source): N/A

**Comment by fntlnz** [comment]
hi @jalseth thanks for opening. I had a similar issue days ago but seems fixed when using a the latest commit from sysdig. Can you try to update sysdig and retry the build?

/triage needs-information

**Comment by fntlnz** [comment]
@jalseth can you post the full command you do for the build? For me it works with something like:

cmake

```bash
docker run -e FALCO_VERSION=0.fntlnz-dev -it --rm --user $(id -u):$(id -g) -v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro -v /etc/passwd:/etc/passwd:ro -e BUILD_BPF=True -e MAKE_JOBS=12 -v ~/Projects/falcosecurity:/source -v ~/Projects/falcosecurity/container-build:/build falcosecurity/falco-builder cmake
```

and then build

```bash
docker run -e FALCO_VERSION=0.fntlnz-dev -it --rm --user $(id -u):$(id -g) -v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro -v /etc/passwd:/etc/passwd:ro -e BUILD_BPF=True -e MAKE_JOBS=12 -v ~/Projects/falcosecurity:/source -v ~/Projects/falcosecurity/container-build:/build falcosecurity/falco-builder bpf
```

**Comment by fntlnz** [comment]
Hi @jalseth any update on this?

**Comment by jalseth** [comment]
Hi @fntlnz, sorry for the delay. I switched the sysdig to the dev git branch and falco to 0.18.0 and still no luck. It says it can't find the kernel build headers, but they're definitely there. Can the CMake build not follow symlinks?

```
$ docker run -e FALCO_VERSION=0.james-dev -it --rm -v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro -e BUILD_BPF=True -e MAKE_JOBS=8 -v $PWD:/source -v $PWD/build:/build falcosecurity/falco-builder cmake
Unable to find image 'falcosecurity/falco-builder:latest' locally
latest: Pulling from falcosecurity/falco-builder
8ba884070f61: Pull complete
4faa8cb317e5: Pull complete
eb6e5ea2d5ea: Pull complete
2bfc9ff060c2: Pull complete
43ea8a02d78b: Pull complete
Digest: sha256:2eeb533c61907f12aa0d039a35ef89b89ec1e64f7b0f5129e4eccd26e992f887
Status: Downloaded newer image for falcosecurity/falco-builder:latest
-- The C compiler identification is GNU 7.3.1
-- The CXX compiler identification is GNU 7.3.1
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Falco version: 0.james-dev
-- Using bundled zlib in '/build/release/zlib-prefix/src/zlib'
-- Using bundled jq in '/build/release/jq-prefix/src/jq'
-- Using bundled nlohmann-json in '/build/release/njson-prefix/src/njson'
-- Using bundled ncurses in '/build/release/ncurses-prefix/src/ncurses'
-- Using bundled b64 in '/build/release/b64-prefix/src/b64'
-- Using bundled yaml-cpp in '/build/release/yamlcpp-prefix/src/yamlcpp'
-- Using bundled openssl in '/build/release/openssl-prefix/src/openssl'
-- Using bundled curl in '/build/release/curl-prefix/src/curl'
-- Using SSL for curl in '--with-ssl=/build/release/openssl-prefix/src/openssl/target'
-- Using bundled LuaJIT in '/build/release/luajit-prefix/src/luajit/src'
-- Using bundled lpeg in '/build/release/lpeg-prefix/src/lpeg'
-- Found autoreconf: /usr/bin
-- Using bundled libyaml in '/build/release/libyaml-prefix/src/libyaml/src'
-- Using bundled lyaml in '/build/release/lyaml-prefix/src/lyaml/ext/yaml'
-- Using bundled tbb in '/build/release/tbb-prefix/src/tbb'
-- Using bundled civetweb in '/build/release/civetweb-prefix/src/civetweb/'
-- Using bundled c-ares in '/build/release/c-ares-prefix/src/c-ares'
-- Using bundled protobuf in '/build/release/protobuf-prefix/src/protobuf'
-- Using bundled grpc in '/build/release/grpc-prefix/src/grpc'
-- Looking for sys/mkdev.h
-- Looking for sys/mkdev.h - not found
-- Looking for sys/sysmacros.h
-- Looking for sys/sysmacros.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /build/release

$ docker run -e FALCO_VERSION=0.james-dev -it --rm -v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro -e BUILD_BPF=True -e MAKE_JOBS=8 -v $PWD:/source -v $PWD/build:/build falcosecurity/falco-builder bpf
Scanning dependencies of target bpf
make[4]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[5]: *** /lib/modules/5.0.0-1025-gcp/build: No such file or directory.  Stop.
make[4]: *** [Makefile:18: all] Error 2
make[3]: *** [driver/bpf/CMakeFiles/bpf.dir/build.make:57: driver/bpf/CMakeFiles/bpf] Error 2
make[2]: *** [CMakeFiles/Makefile2:7203: driver/bpf/CMakeFiles/bpf.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:7215: driver/bpf/CMakeFiles/bpf.dir/rule] Error 2
make: *** [Makefile:2511: bpf] Error 2

$ ls -l /lib/modules/5.0.0-1025-gcp/build
lrwxrwxrwx 1 root root 37 Nov 11 12:52 /lib/modules/5.0.0-1025-gcp/build -> /usr/src/linux-headers-5.0.0-1025-gcp

$ ls -l /lib/modules/5.0.0-1025-gcp/build/
total 1556
lrwxrwxrwx 1 root root      45 Nov 11 12:52 Documentation -> ../linux-gcp-headers-5.0.0-1025/Documentation
lrwxrwxrwx 1 root root      38 Nov 11 12:52 Kbuild -> ../linux-gcp-headers-5.0.0-1025/Kbuild
lrwxrwxrwx 1 root root      39 Nov 11 12:52 Kconfig -> ../linux-gcp-headers-5.0.0-1025/Kconfig
lrwxrwxrwx 1 root root      40 Nov 11 12:52 Makefile -> ../linux-gcp-headers-5.0.0-1025/Makefile
-rw-r--r-- 1 root root 1561925 Nov 11 12:52 Module.symvers
drwxr-xr-x 3 root root    4096 Nov 13 04:37 arch
lrwxrwxrwx 1 root root      37 Nov 11 12:52 block -> ../linux-gcp-headers-5.0.0-1025/block
lrwxrwxrwx 1 root root      37 Nov 11 12:52 certs -> ../linux-gcp-headers-5.0.0-1025/certs
lrwxrwxrwx 1 root root      38 Nov 11 12:52 crypto -> ../linux-gcp-headers-5.0.0-1025/crypto
lrwxrwxrwx 1 root root      39 Nov 11 12:52 drivers -> ../linux-gcp-headers-5.0.0-1025/drivers
lrwxrwxrwx 1 root root      40 Nov 11 12:52 firmware -> ../linux-gcp-headers-5.0.0-1025/firmware
lrwxrwxrwx 1 root root      34 Nov 11 12:52 fs -> ../linux-gcp-headers-5.0.0-1025/fs
drwxr-xr-x 4 root root    4096 Nov 13 04:37 include
lrwxrwxrwx 1 root root      36 Nov 11 12:52 init -> ../linux-gcp-headers-5.0.0-1025/init
lrwxrwxrwx 1 root root      35 Nov 11 12:52 ipc -> ../linux-gcp-headers-5.0.0-1025/ipc
drwxr-xr-x 2 root root    4096 Nov 13 04:37 kernel
lrwxrwxrwx 1 root root      35 Nov 11 12:52 lib -> ../linux-gcp-headers-5.0.0-1025/lib
lrwxrwxrwx 1 root root      34 Nov 11 12:52 mm -> ../linux-gcp-headers-5.0.0-1025/mm
lrwxrwxrwx 1 root root      35 Nov 11 12:52 net -> ../linux-gcp-headers-5.0.0-1025/net
lrwxrwxrwx 1 root root      39 Nov 11 12:52 samples -> ../linux-gcp-headers-5.0.0-1025/samples
drwxr-xr-x 7 root root   12288 Nov 13 04:37 scripts
lrwxrwxrwx 1 root root      40 Nov 11 12:52 security -> ../linux-gcp-headers-5.0.0-1025/security
lrwxrwxrwx 1 root root      37 Nov 11 12:52 sound -> ../linux-gcp-headers-5.0.0-1025/sound
drwxr-xr-x 3 root root    4096 Nov 13 04:37 tools
lrwxrwxrwx 1 root root      38 Nov 11 12:52 ubuntu -> ../linux-gcp-headers-5.0.0-1025/ubuntu
lrwxrwxrwx 1 root root      35 Nov 11 12:52 usr -> ../linux-gcp-headers-5.0.0-1025/usr
lrwxrwxrwx 1 root root      36 Nov 11 12:52 virt -> ../linux-gcp-headers-5.0.0-1025/virt
```

**Comment by fntlnz** [comment]
@jalseth I think that the problem here is that you are sharing the specific kernel modules folder with this part of the command:

```
-v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro
```

Now since that folder in reality is a symlink to the parent directory the build script can’t find that folder, because you didn’t share it, and then it fails.
So I would just share the whole `/lib/modules` folder at this point:

```
-v /lib/modules/$(uname -r):/lib/modules/$(uname -r):ro
```

**Comment by jalseth** [comment]
Figured it out. Since the `/lib/modules/.../build` folder is a symlink to `/usr/src/...` I also needed to share `/usr/src`. However, it looks like the builder container doesn't have `clang` so it can't compile the BPF object.

```
$ docker run -e FALCO_VERSION=0.james-dev -it --rm -v /lib/modules:/lib/modules:ro -e BUILD_BPF=True -e MAKE_JOBS=8 -v $PWD:/source -v $PWD/build:/build -v /usr/src:/usr/src:ro falcosecurity/falco-builder bpf
Scanning dependencies of target bpf
make[4]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[6]: clang: Command not found
make[6]: *** [/source/sysdig/driver/bpf/Makefile:33: /source/sysdig/driver/bpf/probe.o] Error 127
make[5]: *** [Makefile:1614: _module_/source/sysdig/driver/bpf] Error 2
make[4]: *** [Makefile:18: all] Error 2
make[3]: *** [driver/bpf/CMakeFiles/bpf.dir/build.make:57: driver/bpf/CMakeFiles/bpf] Error 2
make[2]: *** [CMakeFiles/Makefile2:7203: driver/bpf/CMakeFiles/bpf.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:7215: driver/bpf/CMakeFiles/bpf.dir/rule] Error 2
make: *** [Makefile:2511: bpf] Error 2
```

**Comment by fntlnz** [comment]
Just for reference, we are having a very similar discussion here https://github.com/falcosecurity/falco/issues/931#issuecomment-559891379

**Comment by fntlnz** [comment]
I just closed #931 in favor of this. We can continue the discussion here since this is older.

In the meanwhile I've been hit by this issue myself but I know that @jalseth is working on a PR to fix - just for context, here's my patch: https://gist.githubusercontent.com/fntlnz/d3bde35336f9bf4d8d7df2a9384400cf/raw/2f5cb9bc2f9577225289771c2c1dba4bc2ad0a44/falco-900.patch

feel free to include anything in it in your PR @jalseth - I know that you have a fix already. Just want to help with more context on how I fixed it.
